src/env.cc

```diff
@@ -0,0 +1,57 @@
+#include <env.h>
+#include <env-inl.h>
+#include <v8.h>
+
+using v8::HandleScope;
+using v8::Local;
+using v8::Message;
+using v8::StackFrame;
+using v8::StackTrace;
+
+namespace node {
+
+void Environment::PrintSyncWarn() const {
+  if (!warn_on_late_sync_)
+    return;
+
+  HandleScope handle_scope(isolate());
+  Local<v8::StackTrace> stack =
+      StackTrace::CurrentStackTrace(isolate(), 10, StackTrace::kDetailed);
+
+  fprintf(stderr, "WARNING: Detected use of sync API\n");
+  for (int i = 0; i < stack->GetFrameCount() - 1; i++) {
+    Local<StackFrame> stack_frame = stack->GetFrame(i);
+    node::Utf8Value fn_name_s(isolate(), stack_frame->GetFunctionName());
+    node::Utf8Value script_name(isolate(), stack_frame->GetScriptName());
+    const int line_number = stack_frame->GetLineNumber();
+    const int column = stack_frame->GetColumn();
+
+    if (stack_frame->IsEval()) {
+      if (stack_frame->GetScriptId() == Message::kNoScriptIdInfo) {
+        fprintf(stderr, "    at [eval]:%i:%i\n", line_number, column);
+      } else {
+        fprintf(stderr,
+                "    at [eval] (%s:%i:%i)\n",
+                *script_name,
+                line_number,
+                column);
+      }
+      fflush(stderr);
+      return;
+    }
+
+    if (strcmp(*fn_name_s, "") == 0) {
+      fprintf(stderr, "    at %s:%i:%i\n", *script_name, line_number, column);
+    } else {
+      fprintf(stderr,
+              "    at %s (%s:%i:%i)\n",
+              *fn_name_s,
+              *script_name,
+              line_number,
+              column);
+    }
+    fflush(stderr);
```

Comment:
What I meant is that it is quite bad to flush stuff on each iteration. It may slow down stuff a lot, depending on the number of frames.